### PR TITLE
feat: support scale's `rangeMin/Max`

### LINKED ISSUE #1721: Support custom domainMin/Max, rangeMin, rangeMax
## Range 

- [ ] Add to schema
- [ ] Make sure to add it to scale supported methods.
- [ ] Modify rangeMixins() in `src/compile/scale/range.ts`
  - [ ] modify the redundant property checker
  - [ ] Support overriding default rangeMin/Max with the specified value. 

## Domain

- [x] add interface for `domainMin` and `domainMax` in scale.ts 
- [x] Modify domain() in `src/compile/scale/domain.ts`
  - [x] Check if the `domainMin/Max` is DateTime object (just like how we check `domain`) and apply `timestamp` if needed.

### PR DESCRIPTION
**Pull request by kanitw**
part of #1721

### CODE REVIEW

**Comment by haldenl** [comment]
oops approved wrong PR

File: build/vega-lite-schema.json [hunk 1]
```diff
@@ -16079,6 +16079,40 @@
...
+            },
+            {
+              "type": "string"
+            },
+            {
+              "items": {
+                "type": "number"
+              },
+              "type": "array"
+            }
```
Comment (haldenl) [inline build/vega-lite-schema.json:16095]: How can rangeMin / rangeMax be an array?

Comment (kanitw) [reply build/vega-lite-schema.json:16095]: Good catch!

Comment (kanitw) [reply build/vega-lite-schema.json:16095]: I simply copied `range`'s type and remove `[]`  😅

Comment (haldenl) [reply build/vega-lite-schema.json]: that was my hunch 😛

**Comment by kanitw** [comment]
Thanks, I'll add a test and merge.

**Pull request merged**